src/output.py

```diff
@@ -94,33 +92,36 @@ def getEditorAndPath():
     return 'vim', 'vim'
 
 
-def getEditFileCommand(filePath, lineNum):
-    editor, _editor_path = getEditorAndPath()
-    if editor in ['vim', 'vim -p'] and lineNum != 0:
-        return '\'%s\' +%d' % (filePath, lineNum)
-    elif editor in ['vi', 'nvim', 'nano', 'joe', 'emacs',
-                    'emacsclient'] and lineNum != 0:
-        return '+%d \'%s\'' % (lineNum, filePath)
-    elif editor in ['subl', 'sublime', 'atom'] and lineNum != 0:
-        return '\'%s:%d\'' % (filePath, lineNum)
-    else:
-        return "'%s'" % filePath
-
-
 def expandPath(filePath):
     # expand ~/ paths
     filePath = os.path.expanduser(filePath)
     # and in case of grep, expand ./ as well
     return os.path.abspath(filePath)
 
 
-def joinEditCommands(partialCommands):
+def joinFilesIntoCommand(filesAndLineNumbers):
     editor, editor_path = getEditorAndPath()
-    if editor in ['vim', 'mvim'] and \
-            not os.environ.get('FPP_DISABLE_SPLIT'):
-        return editor_path + ' -O ' + ' '.join(partialCommands)
-    # Assume that all other editors behave like emacs
-    return editor_path + ' ' + ' '.join(partialCommands)
+    cmd = editor_path + ' '
+    if editor == 'vim -p':
+        firstFilePath, firstLineNum = filesAndLineNumbers[0]
+        cmd += ' +%d %s' % (firstLineNum, firstFilePath)
+        for (filePath, lineNum) in filesAndLineNumbers[1:]:
+            cmd += ' +"tabnew +%d %s"' % (lineNum, filePath)
+    elif editor == 'vim':
+        firstFilePath, firstLineNum = filesAndLineNumbers[0]
+        cmd += ' +%d %s' % (firstLineNum, firstFilePath)
+        for (filePath, lineNum) in filesAndLineNumbers[1:]:
+            cmd += ' +"vsp +%d %s"' % (lineNum, filePath)
```

Comment:
woahhh so `vsp` does vertical split like `-O` and allows you to specify the line numbers? very cool!